File: account_polymorphism_app/src/com/techlabs/account/Account.java
package com.techlabs.account;

public abstract class Account {
	private int accno;
	private String name;
	protected double balance;
	protected final int LOWER_LIMIT = 1000;
	protected final int LOAN_LIMIT = -10000;

	public Account(int accno, String name, double balance) {
		this.accno = accno;
		this.name = name;
		this.balance = balance;
	}
	
	public void deposit(double amount) {
		balance = balance + amount;
	}
	
	public int getAccno() {
		return accno;
	}
	
	public String getName() {
		return name;
	}
	
	public double getBalance() {
		return balance;
	}
	
	public abstract void withdraw(double amount);
}

File: Go/src/account_app/account/account.go
package account

type Account struct {
	accNo   int
	accName string
	balance float64
}

func NewAccount(accNo int, accName string, balance float64) *Account {
	if balance < 0 {
		panic("Balance is less than zero")
	}
	return &Account{
		accNo, accName, balance,
	}
}

func (acc *Account) Deposit(amount float64) {
	acc.balance += amount
}

func (acc *Account) Withdraw(amount float64) {
	if acc.balance-amount < 500 {
		panic("Min balance of 500 should be maintained.")
	}
	acc.balance -= amount
}

func (acc *Account) Balance() float64 {
	return acc.balance
}
func (acc *Account) AccName() string {
	return acc.accName
}
func (acc *Account) AccNo() int {
	return acc.accNo
}
